hammr/commands/image/image.py

```diff
@@ -271,117 +270,142 @@ def help_deploy(self):
-            printer.out("Searching image with id [" + doArgs.id + "] ...")
-            images = self.api.Users(self.login).Images.Getall()
-            images = images.images.image
-            if images is None or len(images) == 0:
-                printer.out("No images available")
-            else:
-                table = Texttable(800)
-                table.set_cols_dtype(["t", "t", "t", "t", "t", "t", "t", "t", "t"])
-                table.header(["Id", "Name", "Version", "Rev.", "Format", "Created", "Size", "Compressed", "Status"])
-                deleteImage = None
-                for image in images:
-                    if str(image.dbId) == str(doArgs.id):
-                        imgStatus = self.get_image_status(image.status)
-                        table.add_row([image.dbId, image.name, image.version, image.revision, image.targetFormat.name,
-                                       image.created.strftime("%Y-%m-%d %H:%M:%S"), size(image.size),
-                                       "X" if image.compress else "", imgStatus])
-                        deleteImage = image
-                if deleteImage is not None:
-                    print table.draw() + "\n"
-                    if generics_utils.query_yes_no(
-                                    "Do you really want to delete image with id " + str(deleteImage.dbId)):
-                        self.api.Users(self.login).Appliances(
-                            generics_utils.extract_id(deleteImage.applianceUri)).Images(deleteImage.dbId).Delete()
-                        printer.out("Image deleted", printer.OK)
-                else:
-                    printer.out("Image not found", printer.ERROR)
+            printer.out("Searching image with id [" + do_args.id + "] ...")
+            images = self.get_all_images()
+            if images is None :
+                raise ValueError("No image found")
+
+            table = Texttable(800)
+            table.set_cols_dtype(["t", "t", "t", "t", "t", "t", "t", "t", "t"])
+            table.header(["Id", "Name", "Version", "Rev.", "Format", "Created", "Size", "Compressed", "Status"])
+            delete_image = None
+            for image in images:
+                if str(image.dbId) == str(do_args.id):
+                    img_status = self.get_image_status(image.status)
+                    table.add_row([image.dbId, image.name, image.version, image.revision, image.targetFormat.name,
+                                   image.created.strftime("%Y-%m-%d %H:%M:%S"), size(image.size),
+                                   "X" if image.compress else "", img_status])
+                    delete_image = image
+            if delete_image is not None:
+                print table.draw() + "\n"
+                if generics_utils.query_yes_no(
+                                "Do you really want to delete image with id " + str(delete_image.dbId)):
+
+                    if is_uri_based_on_appliance(delete_image.uri):
```

Comment:
maybe we can isolate this new code in a "delete_image" or "delete_image_with_uri" method.